contrib/core-contract-tests/contracts/pox/pox-4.clar

```diff
@@ -0,0 +1,1355 @@
+(define-read-only (get-stacking-minimum)
+    (/ stx-liquid-supply STACKING_THRESHOLD_25))
+
+;; Is the address mode valid for a PoX address?
+(define-read-only (check-pox-addr-version (version (buff 1)))
+    (<= (buff-to-uint-be version) MAX_ADDRESS_VERSION))
+
+;; Is this buffer the right length for the given PoX address?
+(define-read-only (check-pox-addr-hashbytes (version (buff 1)) (hashbytes (buff 32)))
+    (if (<= (buff-to-uint-be version) MAX_ADDRESS_VERSION_BUFF_20)
+        (is-eq (len hashbytes) u20)
+        (if (<= (buff-to-uint-be version) MAX_ADDRESS_VERSION_BUFF_32)
+            (is-eq (len hashbytes) u32)
+            false)))
+
+;; Is the given lock period valid?
+(define-read-only (check-pox-lock-period (lock-period uint))
+    (and (>= lock-period MIN_POX_REWARD_CYCLES)
+         (<= lock-period MAX_POX_REWARD_CYCLES)))
+
+;; Is the current burn block height in the prepare phase?
+(define-read-only (check-not-prepare-phase)
+    (> (mod (+ (- burn-block-height (var-get first-burnchain-block-height)) (var-get pox-prepare-cycle-length))
+            (var-get pox-reward-cycle-length))
+        (var-get pox-prepare-cycle-length)))
+
+;; Evaluate if a participant can stack an amount of STX for a given period.
+;; This method is designed as a read-only method so that it can be used as
+;; a set of guard conditions and also as a read-only RPC call that can be
+;; performed beforehand.
+(define-read-only (can-stack-stx (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+                                  (amount-ustx uint)
+                                  (first-reward-cycle uint)
+                                  (num-cycles uint))
+  (begin
+    ;; minimum uSTX must be met
+    (asserts! (<= (get-stacking-minimum) amount-ustx)
+              (err ERR_STACKING_THRESHOLD_NOT_MET))
+
+    (minimal-can-stack-stx pox-addr amount-ustx first-reward-cycle num-cycles)))
+
+;; Evaluate if a participant can stack an amount of STX for a given period.
+;; This method is designed as a read-only method so that it can be used as
+;; a set of guard conditions and also as a read-only RPC call that can be
+;; performed beforehand.
+(define-read-only (minimal-can-stack-stx
+                   (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+                   (amount-ustx uint)
+                   (first-reward-cycle uint)
+                   (num-cycles uint))
+  (begin
+    ;; amount must be valid
+    (asserts! (> amount-ustx u0)
+              (err ERR_STACKING_INVALID_AMOUNT))
+
+    ;; sender principal must not have rejected in this upcoming reward cycle
+    (asserts! (is-none (get-pox-rejection tx-sender first-reward-cycle))
+              (err ERR_STACKING_ALREADY_REJECTED))
+
+    ;; lock period must be in acceptable range.
+    (asserts! (check-pox-lock-period num-cycles)
+              (err ERR_STACKING_INVALID_LOCK_PERIOD))
+
+    ;; stacking must not happen during prepare phase
```

Comment:
Why is this no longer allowed?

Reply:
When stacking during prepare phase the stx are locked but are not earning any rewards. It confuses users and it is extra work for pool operators.

Reply:
Only stx locked before the anchor block are considered for the reward slots.

Reply:
@jcnelson Does it answer you question?

Reply:
I'm sorry for butting in. Even though stackers don't earn rewards, does stacking during the preparation phase confer voting power?

Reply:
My understanding is that they don't get any reward slots. 1 slot, 1 vote on pox. Therfore, no voting right.

Reply:
Yup, that's right. Yes @friedger this answers my question. I didn't know that this was a pain point for users, so I'm glad to see that it's getting fixed :+1: